File: Csharp/Creational/Factory/Show/Factory/ToyFactory.cs
using System;

namespace Factory
{
    public class ToyFactory
    {
        public static IToy CreateToy(string type)
        {
            switch (type)
            {
                case "doll":
                    return new Doll();
                case "car":
                    return new Car();
                case "train":
                    return new Train();
                default:
                    throw new ArgumentException("Invalid toy type");
            }
        }

    }
}

File: TheRayCode/PHP/Creational/Factory/Example2/ToyFactory.php
<?php
include_once 'Car.php';
include_once 'Doll.php';
class ToyFactory
{
    public function createToy($toyType)
    {
        switch ($toyType) {
            case "Car":
                return new Car();
            case "Doll":
                return new Doll();
            default:
                throw new Exception("Invalid toy type");
        }
    }
}